lyrics-core/src/main/java/com/flipkart/lyrics/interfaces/AnnotationSpec.java

```diff
@@ -0,0 +1,56 @@
+package com.flipkart.lyrics.interfaces;
+
+import com.flipkart.lyrics.Song;
+import com.flipkart.lyrics.interfaces.model.NameFormatArgs;
+import com.flipkart.lyrics.interfaces.typenames.ClassName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author kushal.sharma on 10/08/17.
+ */
+public class AnnotationSpec {
+    public final Class<?> clazz;
+    public final ClassName className;
+    public final List<NameFormatArgs> members = new ArrayList<>();
+
+    public AnnotationSpec(Builder builder) {
+        this.clazz = builder.clazz;
+        this.className = builder.className;
+        this.members.addAll(builder.members);
+    }
+
+    public static Builder builder(Class<?> clazz) {
+        return Song.factory.createAnnotationBuilder(clazz);
```

Comment:
create a ClassName out of the Class, will reduce the number of methods to override in the factory